core/base/scalarFieldSmoother/ScalarFieldSmoother.h

```diff
@@ -81,16 +85,26 @@ int ttk::ScalarFieldSmoother::smooth(const triangulationType *triangulation,
   if(!outputData_)
     return -4;
 #endif
-
+  bool useMPI = false;
+#if TTK_ENABLE_MPI
+  if(ttk::isRunningWithMPI() && rankArray != nullptr && globalIds != nullptr)
+    useMPI = true;
+#endif
   SimplexId vertexNumber = triangulation->getNumberOfVertices();
 
   std::vector<dataType> tmpData(vertexNumber * dimensionNumber_, 0);
 
   dataType *outputData = (dataType *)outputData_;
   dataType *inputData = (dataType *)inputData_;
-
+#if TTK_ENABLE_MPI
+  std::unordered_map<SimplexId, SimplexId> gidToLidMap;
```

Comment:
is that required by the `exchangeGhostCells` function?

Reply:
if so, I'd rather move that map to the `AbstractTriangulation` (see lines 3357 to 3396) and provide setters and getters for those. see the functions `preconditionDistributedEdges()`, `getEdgeGlobalId` for examples.

Reply:
Yes, I think moving it the the triangulation is the right call because while there may be some filters not needing a triangulation, I think most which will want to use MPI need a triangulation and then don't need a separate map, I'll check it out.